### ISSUE TITLE
Change Request: skip `sync-client` workflow on forks

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#774]
### Environment

 Not applicable.
This is about the GitHub Actions workflow in this repository, not about running ESLint.

### What problem do you want to solve?

`.github/workflows/sync-client.yml` runs on every push to `main`, including pushes on forks. On a fork the `GH_PAT` secret does not exist, so the job always fails with:

**[error]You must provide either GH_PAT or GH_INSTALLATION_TOKEN**

Example from my fork: https://github.com/tooth-is-silver/npm-eslint-markdown/actions/runs/35310662405

The workflow only makes sense in this repository, because its target is `lumirlumir/lumirlumir` (see `.github/sync-client.yml`). On a fork it can never succeed and just adds a red check on every sync from upstream.

### What do you think is the correct solution?

Add a repository guard to the job so it is skipped on forks:

```yaml
jobs:
  sync-client:
    if: github.repository == 'eslint-markdown/eslint-markdown'
```

For reference, these projects use the same guard to keep a maintainer's personal setup (like sync-client) running only in the maintainer's own repository, not in forks:

- typescript-eslint: https://github.com/typescript-eslint/typescript-eslint/blob/923172a5894be5e83433abc894c35fb777e66344/.github/workflows/release.yml#L63-L64
- eslint-plugin-es-x: https://github.com/eslint-community/eslint-plugin-es-x/blob/39869000a3d63c4c384aea78f19792792f2c5ab7/.github/workflows/Release.yml#L12-L13
- eslint-plugin-promise: https://github.com/eslint-community/eslint-plugin-promise/blob/fb6829b97274d6a3a2313deac4f5a4a3b24544c5/.github/workflows/format.yml#L17

No change in behavior for this repository. Forks get a skipped job instead of a failed one.

### Participation

- [x] I am willing to submit a pull request for this change.

### AI acknowledgment

- [ ] No - I did not use AI to generate this issue report.
- [x] Yes - minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes - substantive assistance (AI-generated non-trivial portions of code, tests, or documentation).

**Comment by lumirlumir** [comment]
I'm 👍 on it. That workflow file is shared across the repositories that I maintain, so I think the following setup would work well in all of my repositories:

```yml
jobs:
  sync-client:
    if: startsWith(github.repository, 'eslint-markdown/') || startsWith(github.repository, 'lumirlumir/')
```
